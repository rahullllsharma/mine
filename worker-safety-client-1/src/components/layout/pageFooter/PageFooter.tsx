import type { PropsWithClassName } from "@/types/Generic";

import cx from "classnames";
import ButtonPrimary from "@/components/shared/button/primary/ButtonPrimary";

export type PageFooterProps = PropsWithClassName<{
  primaryActionLabel?: string;
  onPrimaryClick?: () => void;
  isPrimaryActionDisabled?: boolean;
  isPrimaryActionLoading?: boolean;
}>;

export default function PageFooter({
  className,
  primaryActionLabel,
  onPrimaryClick,
  isPrimaryActionDisabled,
  isPrimaryActionLoading,
}: PageFooterProps): JSX.Element {
  return (
    <footer className={cx("flex items-center p-2 bg-brand-gray-10", className)}>
      <div className="flex flex-1 justify-end">
        {onPrimaryClick && (
          <ButtonPrimary
            onClick={onPrimaryClick}
            label={primaryActionLabel}
            disabled={isPrimaryActionDisabled}
            loading={isPrimaryActionLoading}
          />
        )}
      </div>
    </footer>
  );
}
