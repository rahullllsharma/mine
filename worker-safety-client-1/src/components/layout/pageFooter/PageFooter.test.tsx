import { fireEvent, render, screen } from "@testing-library/react";
import PageFooter from "./PageFooter";

describe(PageFooter.name, () => {
  const mockPrimaryClick = jest.fn();

  it("should be rendered correctly", () => {
    const { asFragment } = render(
      <PageFooter primaryActionLabel="save" onPrimaryClick={mockPrimaryClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should have the "onPrimaryClick" called when the action button is clicked', () => {
    render(
      <PageFooter primaryActionLabel="save" onPrimaryClick={mockPrimaryClick} />
    );
    const buttonElement = screen.getByRole("button", { name: /save/i });
    fireEvent.click(buttonElement);
    expect(mockPrimaryClick).toHaveBeenCalled();
  });

  it('button should be disabled if the "disabled" prop is passed', () => {
    render(
      <PageFooter
        primaryActionLabel="save"
        onPrimaryClick={mockPrimaryClick}
        isPrimaryActionDisabled
      />
    );
    const buttonElement = screen.getByRole("button", { name: /save/i });
    expect(buttonElement).toBeDisabled();
  });
});
