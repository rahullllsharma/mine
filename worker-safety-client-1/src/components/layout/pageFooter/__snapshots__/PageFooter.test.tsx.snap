// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageFooter should be rendered correctly 1`] = `
<DocumentFragment>
  <footer
    class="flex items-center p-2 bg-brand-gray-10"
  >
    <div
      class="flex flex-1 justify-end"
    >
      <button
        class="text-center truncate disabled:opacity-38 disabled:cursor-not-allowed text-base rounded-md font-semibold text-white bg-brand-urbint-40"
        type="button"
      >
        <div
          class="flex items-center justify-center h-full hover:bg-neutral-light-16 focus:bg-neutral-shade-7 active:bg-neutral-shade-7 px-2.5 py-2"
        >
          <span
            class="mx-1 truncate"
          >
            save
          </span>
        </div>
      </button>
    </div>
  </footer>
</DocumentFragment>
`;
